src/domain/checkForAndPlaceOrder.ts

```diff
@@ -424,55 +424,41 @@ export const _printUnfilledOrders = (orders: Map<BytesLike, OrderStatus>) => {
 async function _placeOrder(params: {
   orderUid: string;
   order: any;
-  apiUrl: string;
+  orderBook: OrderBookApi;
   orderRef: string;
   blockTimestamp: number;
+  dryRun: boolean;
 }): Promise<Omit<PollResultSuccess, "order" | "signature"> | PollResultErrors> {
-  const { orderUid, order, apiUrl, orderRef, blockTimestamp } = params;
+  const { orderUid, order, orderBook, orderRef, blockTimestamp, dryRun } =
+    params;
   const logPrefix = `[placeOrder::${orderRef}]`;
   try {
     const postData = {
-      sellToken: order.sellToken,
-      buyToken: order.buyToken,
-      receiver: order.receiver,
+      ...order,
       sellAmount: order.sellAmount.toString(),
       buyAmount: order.buyAmount.toString(),
-      validTo: order.validTo,
-      appData: order.appData,
       feeAmount: order.feeAmount.toString(),
-      kind: order.kind,
-      partiallyFillable: order.partiallyFillable,
-      sellTokenBalance: order.sellTokenBalance,
-      buyTokenBalance: order.buyTokenBalance,
       signingScheme: "eip1271",
-      signature: order.signature,
-      from: order.from,
     };
 
-    // if the apiUrl doesn't contain localhost, post
-    console.log(`${logPrefix} Post order ${orderUid} to ${apiUrl}`);
+    // If the order is a dry run, don't post to the API
+    console.log(
+      `${logPrefix} Post order ${orderUid} to OrderBook on chain ${orderBook.context.chainId}`
+    );
     console.log(`${logPrefix} Order`, postData);
-    if (!apiUrl.includes("localhost")) {
-      const { status, data } = await axios.post(
-        `${apiUrl}/api/v1/orders`,
-        postData,
-        {
-          headers: {
-            "Content-Type": "application/json",
-            accept: "application/json",
-          },
-        }
-      );
-      console.log(`${logPrefix} API response`, { status, data });
+    if (!dryRun) {
+      const orderUid = await orderBook.sendOrder({ ...postData });
+      console.log(`${logPrefix} API response`, { orderUid });
     }
   } catch (error: any) {
     let reasonError = "Error placing order in API";
     if (error.response) {
-      const { status, data } = error.response;
+      const { status } = error.response;
+      const { body } = error;
```

Comment:
Has this been tested? does the SDK return the original response so we can get the status?

Reply:
Yep, this has been tested :+1: